### PR TITLE
Ss/containerizers

### PR DESCRIPTION
**Pull request by sascala**
@nfnt Here's the barest bones of the document. I need to fill this in a bit, but does it look OK to you so far? Next step is a blog post.

### CODE REVIEW

File: 1.8/usage/containerizers.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+post_title: Choosing a Containerizer
+nav_title: Containerizers
+menu_order: 3.5
+---
+
+DC/OS supports both [the default Mesos containerizer and the Docker containerizer](http://mesos.apache.org/documentation/latest/containerizer/).
+
+# Mesos Containerizer
+
+While other containerizers play well with DC/OS, the Mesos containerizer does not depend upon other container technologies and can therefore take advantage of more Mesos features.
```
Comment (sascala) [inline 1.8/usage/containerizers.md]: I'd love to be more specific here without creating a list

File: 1.8/usage/containerizers.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        "cpus": 0.2,
+        "mem": 16.0,
+        "instances": 1
+    }
+	
+For the moment, you can only use these new features of the Mesos containerizer via a JSON app definition, not through the DC/OS web interface.
+
+# Docker Containerizer
+
+Use the Docker containerizer if you need specific features of the Docker package. To specify the Docker containerizer, add the following to your Marathon application definition:
```
Comment (sascala) [inline 1.8/usage/containerizers.md]: such as?

**Comment by sascala** [comment]
Hi, @nfnt . This is the DC/OS-specific page about containerizers. I put the unified containerizer front and center. Is there anything you'd add, correct...?

File: 1.8/usage/containerizers.md (hunk 2 above)
Comment (nfnt) [inline 1.8/usage/containerizers.md]: Mostly you can make use of some docker commands like `docker ps`. And port mappings aren't (yet) available for the Mesos containerizer, though they should land soon.

File: 1.8/usage/containerizers.md (hunk 1 above)
Comment (nfnt) [inline 1.8/usage/containerizers.md]: The [Mesos documentation](http://mesos.apache.org/documentation/latest/container-image/) only talks about why the unification makes sense for Mesos. But at the moment, there's a feature disparity between the containerizers. E.g. GPU support is only available for the Mesos containerizer. Eventually, support for the Docker containerizer will be implemented as well though.

**Comment by nfnt** [comment]
LGTM!

**Comment by sascala** [comment]
@joel-hamill - I'm currently wrestling with getting the local build working, but the text is ready for review.

File: 1.8/usage/containerizers.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
+---
+post_title: Choosing and Using a Containerizer
+nav_title: Containerizers
+menu_order: 3.5
+---
+
+DC/OS supports both [the default Mesos containerizer and the Docker containerizer](http://mesos.apache.org/documentation/latest/containerizer/).
```
Comment (joel-hamill) [inline 1.8/usage/containerizers.md]: I think the intro should be revised to discuss Mesos containerizers at a high level, and then down into the flavors supported. From the standpoint of a newbie "Docker containerizer" doesn't sound Mesos related at all. Is it okay to call it `Mesos Docker containerizer`? E.g. `DC/OS provides support for both the Mesos containerizer and the Mesos Docker containerizer.` 

Also, I think the value proposition is missing from the user perspective - why would they want to know about this feature in the first place? What problem is it solving?

Comment (joel-hamill) [inline 1.8/usage/containerizers.md]: @nfnt what do you think ^

File: 1.8/usage/containerizers.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
+---
+post_title: Choosing and Using a Containerizer
```
Comment (joel-hamill) [inline 1.8/usage/containerizers.md]: `Using Mesos Containerizers`

File: 1.8/usage/containerizers.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+menu_order: 3.5
+---
+
+DC/OS supports both [the default Mesos containerizer and the Docker containerizer](http://mesos.apache.org/documentation/latest/containerizer/).
+
+# Mesos Containerizer
+
+While other containerizers play well with DC/OS, the Mesos containerizer does not depend upon other container technologies and can therefore take advantage of more Mesos features.
+
+The Mesos containerizer now supports provisioning [Docker](https://docker.com/) and [AppC](https://github.com/appc/spec) container images (the ["unified containerizer"](http://mesos.apache.org/documentation/latest/container-image)), so you can take advantage of the Mesos containerizer and still launch an alternative container image.
```
Comment (joel-hamill) [inline 1.8/usage/containerizers.md]: - `The Mesos containerizer supports provisioning...`
- New sentence: `This means that you can use both the Mesos containerizer and other container image types in DC/OS.`

File: 1.8/usage/containerizers.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+This support also removes your dependency on the Docker daemon. The Docker daemon can hang when there are too many containers, making troubleshooting difficult. In addition, Docker must be installed on each of your agent nodes in order to use the Docker containerizer, which means you need to upgrade Docker on the agent nodes each time a new version of Docker comes out. The Mesos containerizer is more stable and built with an eye towards deployment at scale.
```
Comment (joel-hamill) [inline 1.8/usage/containerizers.md]: - `The Mesos containerizer removes your dependency...`
- Break this into 2 sentences: `In addition, Docker must be...`
- This is awkward: `...built with an eye towards...`

File: 1.8/usage/containerizers.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+            },
+            "type": "MESOS"
+        },
+        "args": ["<my-arg>"],
+        "cpus": 0.2,
+        "mem": 16.0,
+        "instances": 1
+    }
+	
+For the moment, you can only use these new features of the Mesos containerizer via a JSON app definition, not through the DC/OS web interface.
```
Comment (joel-hamill) [inline 1.8/usage/containerizers.md]: Reword this to not use the word `new` - this doc will live on long past feature rollout.

File: 1.8/usage/containerizers.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+The Mesos containerizer now supports provisioning [Docker](https://docker.com/) and [AppC](https://github.com/appc/spec) container images (the ["unified containerizer"](http://mesos.apache.org/documentation/latest/container-image)), so you can take advantage of the Mesos containerizer and still launch an alternative container image.
+
+This support also removes your dependency on the Docker daemon. The Docker daemon can hang when there are too many containers, making troubleshooting difficult. In addition, Docker must be installed on each of your agent nodes in order to use the Docker containerizer, which means you need to upgrade Docker on the agent nodes each time a new version of Docker comes out. The Mesos containerizer is more stable and built with an eye towards deployment at scale.
+
+## Provisioning Docker or AppC Containers with the Mesos Containerizer
+
+To run Docker or AppC containers directly from Mesos, specify the container type `MESOS` and a `docker` or `appc` object.
+
+**Note:** This latest version of the Mesos containerizer introduces a new `credential`, with a `principal` and an optional `secret` field to authenticate when downloading the Docker or AppC image.
```
Comment (joel-hamill) [inline 1.8/usage/containerizers.md]: Reword to remove `latest` since this documentation will live on through multiple versions. These kinds of descriptions are better for release notes and marketing collateral.

File: 1.8/usage/containerizers.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        "docker": {
+          "network": "HOST",
+          "image": "<my-image>"
+        },
+        "args": ["<my-arg>"],
+        ]
+      }
+    }
+
+[Learn more about launching Docker containers on Marathon](http://mesosphere.github.io/marathon/docs/native-docker.html).
```
Comment (joel-hamill) [inline 1.8/usage/containerizers.md]: `For more information about launching Docker containers on Marathon, see the [documentation](/docs/1.8/usage/tutorials/docker-app/).` <-- Should we link to our DC/OS doc.

File: 1.8/usage/containerizers.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+While other containerizers play well with DC/OS, the Mesos containerizer does not depend upon other container technologies and can therefore take advantage of more Mesos features.
+
+The Mesos containerizer now supports provisioning [Docker](https://docker.com/) and [AppC](https://github.com/appc/spec) container images (the ["unified containerizer"](http://mesos.apache.org/documentation/latest/container-image)), so you can take advantage of the Mesos containerizer and still launch an alternative container image.
+
+This support also removes your dependency on the Docker daemon. The Docker daemon can hang when there are too many containers, making troubleshooting difficult. In addition, Docker must be installed on each of your agent nodes in order to use the Docker containerizer, which means you need to upgrade Docker on the agent nodes each time a new version of Docker comes out. The Mesos containerizer is more stable and built with an eye towards deployment at scale.
+
+## Provisioning Docker or AppC Containers with the Mesos Containerizer
+
+To run Docker or AppC containers directly from Mesos, specify the container type `MESOS` and a `docker` or `appc` object.
```
Comment (joel-hamill) [inline 1.8/usage/containerizers.md]: I assume this is specified in DC/OS Marathon (aka Services UI or CLI), however this doc sounds like we're specifying directly in Mesos UI/CLI (`To run Docker or AppC containers directly from Mesos...`).

File: 1.8/usage/containerizers.md (hunk 2 above)
Comment (joel-hamill) [inline 1.8/usage/containerizers.md]: @sascala there's some more use case info here: http://mesos.apache.org/documentation/latest/containerizer/#Docker

Comment (joel-hamill) [inline 1.8/usage/containerizers.md]: Can we link to an example Marathon app definition on DC/OS? One that shows how you can launch an app quickly (.e.g. from UI or CLI).

File: 1.8/usage/containerizers.md (hunk 9 above)
Comment (sascala) [inline 1.8/usage/containerizers.md]: The marathon docs go into a fair amount of useful information. I could add a link to the tutorial also.

**Comment by sascala** [comment]
@joel-hamill If this looks good to you I'll publish. I asked Miguel for feedback from a support perspective -- he had one change to make but otherwise said it was good.

**Comment by joel-hamill** [comment]
lgtm, thanks!
